pkg/io/tfio/locals.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package tfio generates a locals.tf file that sets up some of the tf variables
+package tfio
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+
+	"github.com/np-guard/vpc-network-config-synthesis/pkg/ir"
+	"github.com/np-guard/vpc-network-config-synthesis/pkg/utils"
+)
+
+const indentation = "  "
+
+func WriteLocals(defs *ir.ConfigDefs, acl bool) (*bytes.Buffer, error) {
+	data := new(bytes.Buffer)
+	w := bufio.NewWriter(data)
+
+	output := locals(defs, acl)
+	if _, err := w.WriteString(output); err != nil {
+		return nil, err
+	}
+	return data, w.Flush()
+}
+
+func locals(defs *ir.ConfigDefs, acl bool) string {
+	result := "locals {\n"
```

Comment:
How about appending to a slice of strings, then returning `strings.Join(strSlice, '\n')`?